src/java/htsjdk/samtools/SAMBinaryTagAndValue.java

```diff
@@ -73,6 +74,34 @@ private boolean typeSafeEquals(final SAMBinaryTagAndValue that) {
         }
     }
 
+    private boolean valueEquals(SAMBinaryTagAndValue that) {
```

Comment:
I think this update forces and update to `hashCode` as well.  The current hashCode implementation is based on the array identity, not the array values.